Server/src/lobby.erl

```diff
@@ -0,0 +1,115 @@
+			  gen_tcp:send(Socket, io_lib:format("INFO# Wrong password to lobby, ~p", [Lobby_ID]) ++ "~"),
+			  Members
+		  end;
+		found_member -> 
+		  %Tells the user that they already joined this lobby
+		  gen_tcp:send(Socket, io_lib:format("INFO# Already in lobby, ~p", [Lobby_ID]) ++ "~"),
+		  Members
+		end, Diagrams, Lobby_ID);
+
+	%This happens when the message received contains a sequence diagram
+	{create_diagram, Creator_Socket, {Did, Class_names, Classes, Messages}} -> 
+	  %Sends the class names and messages to the client
+      %The character ~ is used as the stop character for when the client should stop reading from the tcp connection
+      Diagram_ID = "l" ++ integer_to_list(Lobby_ID) ++ "d" ++ integer_to_list(Did),
+	  Format_result = io_lib:format("~p", [{Diagram_ID, Class_names}]) ++ "~",
+      send_messages(Members, Format_result),
+	  Self = self(),
+	  loop(Creator_Socket, Password, Members, [{Diagram_ID, Class_names, spawn(fun () -> diagramcoordinator:init(Self, Diagram_ID, {Classes, Messages}) end)}| Diagrams], Lobby_ID);
+	
+	%This case happens when the message recieved contains a command for simulation the diagram
+	{command, Creator_Socket, {Did, Message_request}} -> 
+	  Pid = find_diagram(Did, Diagrams),
+	  %Sends the message to the diagram coordinator.
+	  Pid ! {Message_request, self()},
+	  %Confirmation that the diagram coordinator received the message
+	  receive
+	  	ok -> ok
+	  end,
+	  %Receives the result from the diagram coordinator and sends it to the client
+	  loop(Creator_Socket, Password, Members, Diagrams, Lobby_ID);
+	  
+	%This case happens when a previous message has been readded to the "queue" or if there were to previous messages to step back to.
+	{previous_confirmation, Did, Message} -> 
+	  %Turns the message we want to send into binary.
+	  Format_result = io_lib:format("~p", [{Did, previous_confirmation, Message}]),
+	  %Sends it to the client
+	  send_messages(Members, [Format_result ++ "~"]),
+	  loop(Creator_Socket, Password, Members, Diagrams, Lobby_ID);
+	
+	%This case happens when there is no more messages in the diagram and the user tries to simulate the next message
+	{simulation_done, Did, Message_number} -> 
+	  %Turns the result into binary
+	  Format_result = io_lib:format("~p", [{Did, simulation_finished, Message_number}]),
+      %Sends it to the client
+	  send_messages(Members, [Format_result ++ "~"]),
+	  loop(Creator_Socket, Password, Members, Diagrams, Lobby_ID);
+	  
+	%This case happens when a message has been sent in a diagram and the info is supposed to be sent to the client
+	{message_sent, Did, From, To, Message, Message_number} -> 
+	  %Turns the result into binary
+	  %The character ~ is used as the stop character for when the client should stop reading from the tcp connection
+	  Format_result = io_lib:format("~p", [{Did, From, To, Message, Message_number}]) ++ "~",
+      %Sends it to the client
+	  send_messages(Members, [Format_result]),
```

Comment:
This is inconsistant with all the other send_messages calls. They end with ++ "~" within the function call.